### PR TITLE
first naïve beginnings of a multi-provider for jail

### PR DESCRIPTION
**Pull request by igalic**
this patch is based on #27 and #24.

### CODE REVIEW

**Comment by igalic** [comment]
not quite sure yet why tests are failing…

**Comment by xaque208** [comment]
This is looking good.

File: spec/fixtures/iocage_list-t [hunk 1]
```diff
@@ -0,0 +1,2 @@
+JID   UUID                                  BOOT  STATE  TAG
+-     b439bd7a-5376-11e7-ad91-d979c2a3eb55  off   down   f11-puppet4
```
Comment (xaque208) [inline spec/fixtures/iocage_list-t]: Is this output taken from the new iocage version?  I must have missed the `-t` option.

Comment (igalic) [inline spec/fixtures/iocage_list-t]: nope, that's still the old one

Comment (xaque208) [inline spec/fixtures/iocage_list-t]: Ah yes, hence the name not starting with `py`.

**Comment by igalic** [comment]
aaaaand now, with tests fixed.

**Comment by igalic** [comment]
except rubocop, which i really don't know how to improve at this stage…

**Pull request merged**